src/io/flutter/sdk/FlutterSdkUtil.java

```diff
@@ -94,37 +91,10 @@ private static String sdkRelativePathTo(@NotNull String sdkPath, @NotNull String
     return child.getPath();
   }
 
-  public static boolean isFlutterSdkLibRoot(@Nullable VirtualFile sdk) {
-    if (sdk == null) return false;
-    VirtualFile bin = sdk.findChild("bin");
-    if (bin == null) return false;
-    VirtualFile exec = bin.findChild("flutter");
-    return exec != null;
-  }
-
   public static boolean isFlutterSdkHome(@Nullable String path) {
-    try {
-      return StringUtil.equals(verifyFlutterSdkPath(path), path);
-    }
-    catch (ExecutionException ex) {
-      return false;
-    }
-  }
-
-  @NotNull
-  public static String verifyFlutterSdkPath(@Nullable String path) throws ExecutionException {
-    if (path == null || path.isEmpty()) {
-      throw new ExecutionException(FlutterBundle.message("flutter.sdk.is.not.configured"));
-    }
-    File flutterSdk = new File(path);
-    File bin = new File(flutterSdk, "bin");
-    if (!bin.isDirectory()) {
-      throw new ExecutionException(FlutterBundle.message("flutter.sdk.is.not.configured"));
-    }
-    if (!(new File(bin, "flutter").exists())) {
-      throw new ExecutionException(FlutterBundle.message("flutter.sdk.is.not.configured"));
-    }
-    return path;
+    final File flutterToolFile = new File(path + "/bin/flutter");
+    final File dartLibFolder = new File(path + "/bin/cache/dart-sdk/lib");
```

Comment:
Valid dart SDK is crucial for the suggested model. Under the hood we store 'Dart SDK' global library. Without existing `/bin/cache/dart-sdk/lib` we'll fail to save Dart SDK and thus will loose knowledge about Flutter SDK path.
I think this requirement (to run `flutter doctor` in advance and have valid Dart SDK) is acceptable, as Flutter SDK is unusable without Dart SDK: no analysis, no pub, etc.